license.md

```diff
@@ -60,9 +60,9 @@ You need not contribute prototype changes, extensions, or applications that you
 
 ## Applications
 
-The purpose of this exception to [Copyleft](#copyleft) is to encourage cooperative development of this software, and to prohibit use of this software to create competing substitutes for this software that are not contributed.
+The purpose of this exception to [Copyleft](#copyleft) is to encourage cooperative development of this software, minimizing duplication of effort across competing substitutes.
```

Comment:
The thing I'm nitpicking&mdash;and it's definitely just a nitpick!&mdash;is that the "encouraging cooperation" and the "minimizing duplication of effort" are due to the copyleft terms. This exception is providing permissive terms, and the permissive terms don't encourage cooperation, they encourage use of this software by folks who just want to apply it to their proprietary application, right?

I saw elsewhere that you're trying to avoid the term "proprietary", which makes sense, I guess it's kind of a Stallmanism; another wording I was thinking of was "reserving rights". As in, "The purpose of this exception to Copyleft is to allow you to reserve all rights to applications that you build with the software, so long as they are not competing substitutes that undermine cooperative development of this software."

If you want an exception section to be positively encouraging cooperative development, perhaps it would make more sense to structure the license as a whole to be permissive-with-exceptions rather than copyleft-with-exceptions?

Reply:
@laughinghan You're right about the purpose of the exception versus the purpose of the copyleft scheme overall. That occurred to me, too. I should have known you'd go right to it!

I _am_ avoiding "proprietary", but not for any hidden reason. Open licensing wonks use that term in a way totally peculiar to their tribe: "not open". Lawyers don't understand it the same way, or even use it much in this context, for any meaning. So we can't rely on it for any specific content without calling that kind of usage out. Which doesn't seem worth the trouble.

I hear you on permissive-with-exceptions. But I think there are good reasons we tend to see copyleft-with-exceptions instead. It's really hard to talk about that generally, without referencing specific licenses. But I think the most important consideration is what happens when terms aren't crystal clear about whether copyleft triggers in a particular situation. Starting with "contribute back" as the general rule, and carving out exceptions from there, helps to defend the general thrust toward requiring contribution. It's fail-open, rather than fail-closed, so to speak.

Reply:
Addressed in 056266b67e09e5ada222e56b832a52d6d6c95ad0